### PR TITLE
[11.0][MIG] account_payment_order

### PR DESCRIPTION
**Pull request by etobella**
Migration to 11.0
Still working on tests and multicompany environments

Depends on

- [x] account_payment_mode #399 
- [x] account_payment_partner #400

### CODE REVIEW

File: account_payment_order/README.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+* account_payment_partner
+* base_iban
+* document
+
+This modules is part of the OCA/bank-payment suite.
+
+Configuration
+=============
+
+This module adds several options on Payment Modes, cf Accounting > Configuration > Management > Payment Modes.
```
Comment (rgarnau) [inline account_payment_order/README.rst]: 'Invoicing' instead of 'Accounting'

File: account_payment_order/README.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Usage
+=====
+
+You can create a Payment Order via the menu Accounting > Payments > Payment Orders and then select the move lines to pay.
```
Comment (rgarnau) [inline account_payment_order/README.rst]: Invoicing

File: account_payment_order/README.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+You can create a Debit Order via the menu Accounting > Payments > Debit Orders and then select the move lines to debit.
```
Comment (rgarnau) [inline account_payment_order/README.rst]: Invoicing

File: account_payment_order/README.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+This module also adds a button *Add to Payment Order* on supplier invoices and a button *Add to Debit Order* on customer invoices.
+
+You can print a Payment Order via the menu Accounting > Payments > Payment Orders and then select the payment oder to print.
```
Comment (rgarnau) [inline account_payment_order/README.rst]: Invoicing

**Review by rgarnau — COMMENTED** [review]
LGTM :+1: (minor comments inline)

**Review by rgarnau — APPROVED** [review]

File: account_payment_order/wizard/account_payment_line_create.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                #  '|',
+                ('account_id.internal_type', '=', 'payable'),
+                #  '&',
+                #  ('account_id.internal_type', '=', 'receivable'),
+                #  ('reconcile_partial_id', '=', False),  # TODO uncomment
+            ]
+        elif self.order_id.payment_type == 'inbound':
+            domain += [
+                ('debit', '>', 0),
+                ('account_id.internal_type', '=', 'receivable'),
```
Comment (JordiBForgeFlow) [inline account_payment_order/wizard/account_payment_line_create.py]: My concern with payment.orders is that it does not seem to be able to handle refunds. So, in the scenario where you have an angry customer that expects a refund (and is not likely to come again - so little chance to reconcile the refund in the future with a new invoice), you cannot create a payment order, due to this added domain. 

IMHO what should only matter here is that with inbound payment orders you pay debit > 0, for receivables and payables, and with outbound payment orders you pay credit > 0, for receiv & payab.

cc @alexis-via @pedrobaeza @etobella

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Technical and functional review.

File: account_payment_order/README.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+Usage
+=====
+
+You can create a Payment Order via the menu Invoicing/Accounting > Payments > Payment Orders and then select the move lines to pay.
+
+You can create a Debit Order via the menu Invoicing/Accounting > Payments > Debit Orders and then select the move lines to debit.
+
+This module also adds a button *Add to Payment Order* on supplier invoices and a button *Add to Debit Order* on customer invoices.
+
+You can print a Payment Order via the menu Invoicing/Accounting > Payments > Payment Orders and then select the payment oder to print.
```
Comment (HviorForgeFlow) [inline account_payment_order/README.rst]: order

File: account_payment_order/models/account_invoice.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+                    "The invoice %s is not in Open state") % inv.number)
+            if not inv.payment_mode_id:
+                raise UserError(_(
+                    "No Payment Mode on invoice %s") % inv.number)
+            if not inv.move_id:
+                raise UserError(_(
+                    "No Journal Entry on invoice %s") % inv.number)
+            if not inv.payment_order_ok:
+                raise UserError(_(
+                    "The invoice %s has a payment mode '%s' "
```
Comment (HviorForgeFlow) [inline account_payment_order/models/account_invoice.py:58]: no string formatting:
"which is not selectable in payment orders.") % (inv.number, inv.payment_mode_id.display_name))

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by angelmoya** [comment]
:+1:

**Pull request merged**